### PR TITLE
[bitnami/thanos] use the `tpl` function in querier of helm3

### PR DESCRIPTION
**Pull request by Cellebyte**
* this better integrates with umbrella helm charts.
* It is now possible to setup templates in `values.yaml`

**Description of the change**

* It enabled templating inside of 
```yaml
querier:
  enabled: true
  logLevel: info
  replicaLabel: replica
  dnsDiscovery:
    enabled: true
    sidecarsService: "{{ .Release.Name }}-prometheus-thanos"
    sidecarsNamespace: "{{ .Release.Namespace }}"
```

**Benefits**

**Possible drawbacks**
* If you add `{{}}` inside the two strings it will template those brackets.

**Applicable issues**
* Dont' know what will happen, if you set a dictionary or list instead of a template string.
* A normal string should render just fine.

  - fixes #

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Comment by Cellebyte** [comment]
Please merge #2020 before #2024 . 
I think I need to resolve the merge issues before this one can be merged.

File: bitnami/thanos/templates/querier/deployment.yaml [hunk 1]
```diff
@@ -62,7 +62,7 @@ spec:
             - --store.sd-files=/conf/servicediscovery.yml
             {{- end }}
             {{- if and .Values.querier.dnsDiscovery.enabled .Values.querier.dnsDiscovery.sidecarsService .Values.querier.dnsDiscovery.sidecarsNamespace }}
-            - --store=dnssrv+_grpc._tcp.{{ .Values.querier.dnsDiscovery.sidecarsService }}.{{ .Values.querier.dnsDiscovery.sidecarsNamespace }}.svc.{{ .Values.clusterDomain }}
+            - --store=dnssrv+_grpc._tcp.{{ tpl .Values.querier.dnsDiscovery.sidecarsService . }}.{{ tpl .Values.querier.dnsDiscovery.sidecarsNamespace . }}.svc.{{ .Values.clusterDomain }}
```
Comment (dani8art) [inline bitnami/thanos/templates/querier/deployment.yaml]: could you follow the same approach as we follow on https://github.com/bitnami/charts/blob/master/bitnami/thanos/values.yaml#L256 using https://github.com/bitnami/charts/blob/master/bitnami/thanos/templates/_helpers.tpl#L328

File: bitnami/thanos/Chart.yaml [hunk 2]
```diff
@@ -15,4 +15,4 @@ maintainers:
 name: thanos
 sources:
   - https://github.com/bitnami/bitnami-docker-thanos
-version: 0.1.8
+version: 0.1.9
```
Comment (dani8art) [inline bitnami/thanos/Chart.yaml]: I think these changes would be a minor.

File: bitnami/thanos/values.yaml [hunk 3]
```diff
@@ -61,16 +61,17 @@ querier:
   ##
   replicaLabel: replica
   
-  ## Dinamically configure store APIs using DNS discovery
+  ## Dynamically configure store APIs using DNS discovery
```
Comment (dani8art) [inline bitnami/thanos/values.yaml]: Please, could you add a small phrase to the `README.md` at the values section explaining how it is going to work

File: bitnami/thanos/templates/querier/deployment.yaml (hunk 1 above)
Comment (Cellebyte) [reply bitnami/thanos/templates/querier/deployment.yaml]: ok, i will use that template function, thanks for guidance ;)

File: bitnami/thanos/Chart.yaml (hunk 2 above)
Comment (Cellebyte) [reply bitnami/thanos/Chart.yaml]: will bump it to `0.2.0`

File: bitnami/thanos/values.yaml (hunk 3 above)
Comment (Cellebyte) [reply bitnami/thanos/values.yaml]: Will do that. Thx

**Comment by Cellebyte** [comment]
@dani8art 

`- --store=dnssrv+_grpc._tcp.RELEASE-NAME-prometheus-thanos.default.svc.cluster.local`
The template is working as expected thank you for your hint with the tplValue function in _helperts.tpl.

Appreciated it.

**Comment by Cellebyte** [comment]
> Please, could you resolve conflict and add the documentation of this new functionality to README.md in values section?

@dani8art I added the little comment.
```yaml
    ## Evaluated as a template.
```

**Pull request merged**